### PR TITLE
feat: implement @composeDirective

### PR DESCRIPTION
**Pull request by Aenimus**
## Summary by CodeRabbit

* **New Features**
  * Directive composition/linking with enhanced validation and link/import parsing
  * BatchNormalizer for coordinated multi-subgraph batch normalization and federation-aware normalization

* **Bug Fixes**
  * Improved and structured error/warning reporting for directive misuse, imports, and subgraph naming
  * Stronger validation of directive arguments, locations, and default values

* **Refactor**
  * Reworked directive handling to a federated model and tightened public typings

* **Tests**
  * Added tests covering composed directives, federation propagation, and override behavior

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/docs-website](https://github.com/wundergraph/docs-website).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

## Open Source AI Manifesto

This project follows the principles of the [Open Source AI Manifesto](https://human-oss.dev). Please ensure your contribution aligns with its principles.

### CODE REVIEW

File: composition/src/v1/normalization/utils.ts [hunk 1]
```diff
@@ -459,10 +494,310 @@ export function initializeDirectiveDefinitionDatas(): Map<string, DirectiveDefin
...
+      existingDataByName.set(directiveName, directiveData);
+      executableDirectiveDatasByName.delete(directiveName);
+      continue;
+    }
+
+    if (existingData.version < directiveData.version) {
+      // Only propagate the latest version's description if any
+      existingData.description = directiveData.description;
+      existingDataByName.set(directiveName, directiveData);
+    }
```
[the remark this answers is not in the document]
Comment (Aenimus) [reply composition/src/v1/normalization/utils.ts]: This is intentional. Added a comment.

**Review by wilsonrivera — APPROVED** [review]
LGTM

**Review by endigma — APPROVED** [review]
